### PR TITLE
T6703: Adds option to configure AMD pstate driver

### PR DESCRIPTION
**Pull request by nvollmar**
## Change Summary
Add system option to configure AMD pstate driver

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
https://vyos.dev/T6703

## Related PR(s)
vyos/vyos-build#755
vyos/vyos-documentation#1543

## Component(s) name
system

## Proposed changes
Enable system command to set kernel boot args for configuring AMD pstate driver.

## How to test
```
set system option kernel amd-pstate-driver passive
```

## Smoketest result
None

## Checklist:
- [X] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [X] I have linked this PR to one or more Phabricator Task(s)
- [X] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [X] My commit headlines contain a valid Task id
- [X] My change requires a change to the documentation
- [X] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/system_option.xml.in [hunk 1]
```diff
@@ -49,6 +49,26 @@
...
               </leafNode>
+              <leafNode name="amd-pstate-driver">
+                <properties>
+                  <help>Enables and configures pstate driver for AMD Ryzen and Epyc CPUs</help>
+                  <completionHelp>
+                    <list>active passive guided</list>
+                  </completionHelp>
+                  <valueHelp>
+                    <format>active</format>
+                    <description>This is the low-level firmware control mode based on the profile set and the system governor has no effect</description>
```
Comment (dmbaturin) [inline interface-definitions/system_option.xml.in]: I think these descriptions are too wordy.
```suggestion
                    <description>Give all control to the pstate driver instead of the system governor</description>
```

File: interface-definitions/system_option.xml.in [hunk 2]
```diff
@@ -49,6 +49,26 @@
...
+                  </valueHelp>
+                  <valueHelp>
+                    <format>passive</format>
+                    <description>The driver allows the system governor to manage CPU frequency while providing available performance states</description>
```
Comment (dmbaturin) [inline interface-definitions/system_option.xml.in]:
```suggestion
                    <description>Allow the system governor to manage the CPU</description>
```

File: interface-definitions/system_option.xml.in [hunk 3]
```diff
@@ -49,6 +49,26 @@
...
+                  </valueHelp>
+                  <valueHelp>
+                    <format>guided</format>
+                    <description>The driver allows to set desired performance levels and the firmware selects a performance level in this range and fitting to the current workload</description>
```
Comment (dmbaturin) [inline interface-definitions/system_option.xml.in]:
```suggestion
                    <description>Allow the firmware to select performance levels based on the workload</description>
```

**Review by dmbaturin — CHANGES_REQUESTED** [review]
It think help strings can be improved.

File: interface-definitions/system_option.xml.in (hunk 1 above)
Comment (nvollmar) [reply interface-definitions/system_option.xml.in]: This is not quite correct as it's not the p-state driver which is in control. I'll try to reword it.

**Comment by nvollmar** [comment]
@dmbaturin I've updated/shortened the wording based on your proposals.

**Review by dmbaturin — APPROVED** [review]

**Review by c-po — APPROVED** [review]

**Comment by c-po** [comment]
@MergifyIo backport circinus
